### ISSUE TITLE
Arrow key navigation in table block

### ISSUE THREAD

**Issue by nosolosw** [WordPress/gutenberg#14675]
Spun off from https://github.com/WordPress/gutenberg/pull/14462#issuecomment-476165700

When the caret is at the table block:

- the arrow down should move to the cell below instead of to the right cell
- the arrow up key should move the caret to the cell above instead of the left cell

**Comment by ellatrix** [comment]
Currently, this roughly how the logic is:

1. We look for the closest tabbable element in a given direction.
2. We place the caret at the closet `y` position.

The problem is (1): DOM positioning may not be just vertical, but also horizontal, as is the case in a table block. Similar to #14462, we come to the same conclusion: we're looking for a visual position, but we're checking DOM position.

The solution is to find the closest element vertically that can receive the caret in the closest block. If there is no such element, the block should be selected.

Practically this probably means that we have to test if the browser can set a range at any `x` position with that fixed `y` position, perhaps skipping a few `x` positions at a time.

**Comment by tellthemachines** [comment]
Noting here, as per the issue linked above, that currently it's impossible to move focus with the left and right arrow keys to a cell that's hidden from view. So if we have a wide table with horizontal scroll:

<img width="802" alt="Screen Shot 2020-04-20 at 2 08 57 pm" src="https://user-images.githubusercontent.com/8096000/79713959-b5be0280-8312-11ea-8034-0ba7c94acdd5.png">

with the cursor at the "basil" column I am not able, by using the right arrow, to navigate to the next column. If I scroll the hidden column into view, I am able to focus it normally.

**Comment by ellatrix** [comment]
@tellthemachines could you open a new issue for that? It seems unrelated.

**Comment by ellatrix** [comment]
The up/down arrow issue should be fixed in #22105.
